File: FlutterCompareApplication/flutter_compare/lib/barrierefreiheit.dart
import 'package:flutter/material.dart';

class BFPage extends StatefulWidget {

  @override
  _BF createState() => _BF();
}

class _BF extends State<BFPage> {

  double size = 14;


  changeTarget(String c) {
    
    if (c == '-') {
      setState(() {
        size = size - 1;
      });
    } else {
      setState(() {
        size = size + 1;
      });
    }
    
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: <Widget>[
          Semantics(
            //button: true,
            child: RaisedButton(onPressed: () => {}, child: 
              Semantics(
                child: Text('Button BF Text'),
                hint: 'Ein Hilfetext für die Barrierefreiheit des Buttons',
                label: 'Barrierefreiheit Button',
              )
            ,),
          ),
          Semantics(
            child: Text('Label BF Text'),
            hint: 'Ein Hilfetext für die Barrierefreiheit des Textes',
            label: 'Barrierefreiheit Label',
          ),
          RaisedButton(onPressed: () {
            changeTarget('+');
          }, child: Text('Größer'),),
          RaisedButton(onPressed: () {
            changeTarget('-');
          }, child: Text('Kleiner'),),
          Text('Skalier mich', style: new TextStyle(fontSize: size),)
        ],
      ),
      backgroundColor: Colors.lightBlue,
    );
  }
}

File: ReactNativeCompareApp/ReactNativeCompareApp/Barrierefreiheit.js
import React from 'react';
import {
  StyleSheet,
  View,
  Button,
  Text,
  Image,
  TouchableOpacity
} from 'react-native';

class Barrierefreiheit extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      fontSizeOfText: 14
    };
  }

  scaleText(sign) {
    if (sign === '+') {
      this.setState({
        fontSizeOfText: (this.state.fontSizeOfText + 1)
      });
    } else if (sign === '-') {
      this.setState({
        fontSizeOfText: (this.state.fontSizeOfText - 1)
      });
    }
  }

  render() {
    return (
      <View>
        <Button
          title="Button BF Text"
          accessible={true}
          accessibilityLabel="Barrierefreiheit Button"
          accessibilityHint="Ein Hilfetext für die Barrierefreiheit des Buttons"
        />
        <Text
          accessible={true}
          accessibilityLabel="Barrierefreiheit Label"
          accessibilityHint="Ein Hilfetext für die Barrierefreiheit des Textes"
        >
          Label BF Text
        </Text>
        <Button title="Größer" onPress={() => this.scaleText('+')} />
        <Button title="Kleiner" onPress={() => this.scaleText('-')} />
        <Text style={{ fontSize: this.state.fontSizeOfText }}>
          Skalier mich
        </Text>
      </View>
    );
  }
}

export default Barrierefreiheit;
